### ISSUE TITLE
$translate.use should be able to fail/try again if failed once while loading a file

### ISSUE THREAD

**Issue by JamesBoon** [angular-translate/angular-translate#969]
Hi,

if you are using the `$translateStaticFilesLoader` with `$translate.use(key)` and get a 404 once, you can never load the file again. The returned promise will always be resolved before the actual file is re-requested.
This affects all versions.

To fix it, `langPromises[key]` must be cleared after `$translate.use(key)` failed.
Adding `langPromises[key] = undefined;` before [emitting the change error](https://github.com/angular-translate/angular-translate/blob/canary/src/service/translate.js#L1543) should fix it.

Here is a test which will fail before the fix:
```javascript
var $translate, $httpBackend, $translateStaticFilesLoader;

beforeEach(module('pascalprecht.translate', function ($translateProvider) {
  $translateProvider.useStaticFilesLoader({prefix : 'lang_', suffix : '.json'});
}));

beforeEach(inject(function (_$translate_, _$httpBackend_, _$translateStaticFilesLoader_) {
  $httpBackend = _$httpBackend_;
  $translate = _$translate_;
  $translateStaticFilesLoader = _$translateStaticFilesLoader_;
  $httpBackend.when('GET', 'lang_nt_VD.json').respond(404);
}));

afterEach(function() {
  $httpBackend.verifyNoOutstandingExpectation();
  $httpBackend.verifyNoOutstandingRequest();
});    

it('try load the file again', function () {
  var wasResolved = false,
	wasRejected = false,
	promise = $translate.use('nt_VD');
  promise.then(resolved, rejected);
  $httpBackend.flush();

  expect(wasResolved).toEqual(false);
  expect(wasRejected).toEqual(true);

  promise = $translate.use('nt_VD');
  promise.then(resolved, rejected);
  $httpBackend.flush();

  expect(wasResolved).toEqual(false);
  expect(wasRejected).toEqual(true);

  function resolved() {
	wasResolved = true;
  }
  function rejected() {
	wasRejected = true;
  }
});
```

**Comment by knalli** [comment]
fyi: I've made a proposal already in 9f1b0d1 related to #824

**Comment by JamesBoon** [comment]
Hi @knalli,
thank you for the info.
I have created some tests for this and had some problems with this proposal. The PR #970 fixes this. Please have a look and tell me what you think.

**Comment by JamesBoon** [comment]
Closed by #970
